scylla-cdc-lib/src/test/java/com/scylladb/cdc/lib/AlterTableIT.java

```diff
@@ -167,4 +178,193 @@ private synchronized void printDetails(RawChange change) {
+      firstChange = new AtomicReference<>(null);
+      firstChangeAfterFetchingNewPage = new AtomicReference<>(null);
+      RawChangeConsumer changeConsumer =
+          change -> {
+            lastConsumedTime.set(System.currentTimeMillis());
+
+              firstChange.compareAndSet(null, change);
+              if (unlockedNextPageLock.get()) {
+                consumedChangeAfterAlter.set(true);
+                firstChangeAfterFetchingNewPage.compareAndSet(null, change);
+                verifySchemaAfterAlter(change);
+              } else {
+                verifySchemaBeforeAlter(change);
+              }
+              String toString = change.toString(); // forces Driver3RawChange to go through all ColumnDefinitions
+
+            return CompletableFuture.completedFuture(null);
+          };
+
+      try (CDCConsumer consumer =
+               CDCConsumer.builder()
+                   .addContactPoint(new InetSocketAddress(hostname, port))
+                   .addTable(new TableName(keyspace, table))
+                   .withConsumer(changeConsumer)
+                   .withQueryTimeWindowSizeMs(15 * 1000)
+                   .withConfidenceWindowSizeMs(confidenceWindowSeconds * 1000)
+                   .withWorkersCount(1)
+                   .withQueryOptionsFetchSize(1)
+                   .build()) {
+        ReentrantReadWriteLock nextPageLock = null;
+        try {
+          // Access the LocalTransport field in CDCConsumer
+          java.lang.reflect.Field transportField = CDCConsumer.class.getDeclaredField("transport");
+          transportField.setAccessible(true);
+          LocalTransport transport = (LocalTransport) transportField.get(consumer);
+
+          // Access the WorkerConfiguration.Builder field in LocalTransport
+          java.lang.reflect.Field builderField = LocalTransport.class.getDeclaredField("workerConfigurationBuilder");
+          builderField.setAccessible(true);
+          WorkerConfiguration.Builder builder = (WorkerConfiguration.Builder) builderField.get(transport);
+
+          // Access the session field in CDCConsumer
+          java.lang.reflect.Field sessionField = CDCConsumer.class.getDeclaredField("session");
+          sessionField.setAccessible(true);
+          Driver3Session driver3Session = (Driver3Session) sessionField.get(consumer);
+
+          // Create MockDriver3WorkerCQL and replace the cql field in the builder
+          MockDriver3WorkerCQL mockWorkerCQL = new MockDriver3WorkerCQL(driver3Session);
+          // Grab the lock to control it later on
+          nextPageLock = mockWorkerCQL.nextPageLock;
+          java.lang.reflect.Field cqlField = WorkerConfiguration.Builder.class.getDeclaredField("cql");
+          cqlField.setAccessible(true);
+          cqlField.set(builder, mockWorkerCQL);
+        } catch (Exception e) {
+          throw new RuntimeException("Failed to replace Driver3WorkerCQL with MockDriver3WorkerCQL", e);
+        }
```

Comment:
Can you please find better way to do that, without hacking into instances.
If necessary add additional API to builders.

Reply:
It will require modifying the accesses in production code, but I can.